static/src/js/components/Legend/Legend.js

```diff
@@ -75,12 +78,23 @@ export const Legend = ({
   }, [barRef.current, colorMap])
 
   // If no colormap is available, hide the legend.
-  if (!colorMap || !colorMap.scale) return null
-
-  const { scale = {} } = colorMap
-  const { labels = [], colors = [] } = scale
-  const minLabel = labels[0]
-  const maxLabel = labels[labels.length - 1]
+  if (!colorMap || (!colorMap.scale && !colorMap.classes)) return null
+
+  const { scale = {}, classes = {} } = colorMap
+  let { colors = [], labels = [] } = scale
+  let minLabel
+  let maxLabel
+
+  const qualitative = Object.keys(classes).length > 0
+  const hoverPrompt = 'Hover for class names'
+  if (qualitative) {
+    ({ colors = [], labels = [] } = classes)
+    minLabel = hoverPrompt
+    maxLabel = null
+  } else {
+    [minLabel] = labels
```

Comment:
```suggestion
    minLabel = labels[0]
```
nitpick; I know what you currently have does the same thing this is just a preference I have in this case to match the line below more clearly

Reply:
Oh I totally agree. This is one of the sillier ESLint rules we might consider disabling globally.

Reply:
That might be an issue on your configuration in vscode.. eslint isn't complaining if you do it either way, otherwise this ci build would fail.

Reply:
I'll need to figure out why the VSCode ESLint plugin isn't using the same configuration as the project. I'll push a fix to switch it to the more readable version.

Reply:
ESLint failed after pushing the change

Reply:
I see I was wrong. Its because the var isn't a const. I'll dismiss this. just use the eslint preference